### PR TITLE
[MI-1853] refactor notification; added notification when order is not found

### PR DESCRIPTION
**Pull request by kcasas**
/cc @zendesk/mintegrations

### Description
* refactored error notification to use a more generic and reusable notification system
* added notification message when order id filter did not match any orders.

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-1853

### Risks
* [low] error notification might not work properly

### CODE REVIEW

File: src/Sagas/orders.js [hunk 1]
```diff
@@ -18,6 +19,11 @@ export function* fetchOrder(action) {
       if (order) {
         response = yield call(callApi, `/admin/orders/${order.id}.json`);
       } else {
+        yield put(addNotification(
+          'warning',
+          'orders',
+          { message: `Order with id #${action.orderId} not found` },
```
Comment (miogalang) [inline src/Sagas/orders.js]: Is this string translated?

File: tests/Middlewares/CustomerResponseMiddleware.test.js [hunk 2]
```diff
@@ -54,9 +54,10 @@ describe('CustomerResponseMiddleware', () => {
     customerResponseMiddleware(store)(next)(action);
     expect(store.dispatch.args[0][0]).toEqual({ type: FETCH_CUSTOMER_FAILED });
     expect(next.args[0][0]).toEqual({
-      type: ADD_ERROR,
-      index: 'customer',
-      error: { message: 'Customer could not be found.' },
+      type: ADD_NOTIFICATION,
+      key: 'customer',
+      kind: 'errors',
+      data: { message: 'Customer could not be found.' },
```
Comment (miogalang) [inline tests/Middlewares/CustomerResponseMiddleware.test.js]: Same comment about i18n

File: tests/Reducers/NotificationReducer.test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import faker from 'faker';
+import { Reducer } from 'redux-testkit';
+import notificationReducer from '../../src/Reducers/NotificationReducer';
+import {
+  ADD_NOTIFICATION,
+  REMOVE_NOTIFICATION,
+} from '../../src/Actions/constants.js';
+
+describe('error reducer', () => {
+  it('should add an error to the tree based on kind and key', () => {
```
Comment (samgavinio) [inline tests/Reducers/NotificationReducer.test.js]: should add *a notification* to the tree based on kind and key

File: tests/Reducers/NotificationReducer.test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import faker from 'faker';
+import { Reducer } from 'redux-testkit';
+import notificationReducer from '../../src/Reducers/NotificationReducer';
+import {
+  ADD_NOTIFICATION,
+  REMOVE_NOTIFICATION,
+} from '../../src/Actions/constants.js';
+
+describe('error reducer', () => {
```
Comment (samgavinio) [inline tests/Reducers/NotificationReducer.test.js]: Notification reducer now?

File: tests/Reducers/NotificationReducer.test.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    const key2 = faker.random.word();
+    action.key = key2;
+    Reducer(notificationReducer).withState(expectedState).expect(action).toReturnState({
+      foo: 'bar',
+      [kind]: { [key]: data, [key2]: data },
+    });
+  });
+
+  it('should remove an error from the tree based on key', () => {
```
Comment (samgavinio) [inline tests/Reducers/NotificationReducer.test.js]: should remove *a notification* from the tree based on key

File: tests/Reducers/NotificationReducer.test.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    const kind = faker.random.word();
+    const key = faker.random.word();
+    const action = { type: REMOVE_NOTIFICATION, kind, key };
+
+    Reducer(notificationReducer).withState({ [kind]: { [key]: 'foo' }, foo: 'bar' }).expect(action).toReturnState({
+      [kind]: {},
+      foo: 'bar',
+    });
+  });
+});
```
Comment (samgavinio) [inline tests/Reducers/NotificationReducer.test.js]: Also lost the tests for `addError` and `removeError`, we probably should keep those.

File: src/Sagas/orders.js (hunk 1 above)
Comment (kcasas) [reply src/Sagas/orders.js]: Due to the limitation of the current translation implementation, a refactor is needed. I've created a split task https://zendesk.atlassian.net/browse/MI-1866 for this.

**Pull request merged**